.github/scripts/bot-advanced-check.sh

```
@@ -1,93 +1,231 @@
+# Helper: has bot already commented for user?
+#######################################
+already_commented() {
+  local user=$1
+  local marker="$COMMENT_MARKER_PREFIX @$user"
+
+  gh issue view "$ISSUE_NUMBER" --repo "$REPO" \
+    --json comments \
+    --jq --arg marker "$marker" '
+      .comments[].body
+      | select(contains($marker))
+    ' | grep -q .
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)